Source/DafnyCompilers/AST.dfy

```diff
@@ -1,350 +1,350 @@
-              // For instance, maps can contain keys which don't have a decidable equality,
-              // and sequences can contain elements which also don't have a decidable equality
-              // (in which case we don't have a decidable equality over the sequences, but it
-              // is fine).
-              && eltType.NoLeftFunction()
-              && match kind {
-                case Seq => true
-                case Set => eltType.NoLeftFunction()
-                case Multiset => eltType.NoLeftFunction()
-                case Map(kt) => kt.WellFormed() && kt.NoLeftFunction()
-              }
-            case Function(args: seq<Type>, ret: Type) =>
-              && (forall i | 0 <= i < |args| :: args[i].WellFormed())
-              && ret.WellFormed()
-            case Class(classType: ClassType) =>
-              && (forall i | 0 <= i < |classType.typeArgs| :: classType.typeArgs[i].WellFormed())
-          }
+      // TODO: remove?
+      predicate method WellFormed() {
+        match this {
+          case Unit => true
+          case Bool => true
+          case Char => true
+          case Int => true
+          case Real => true
+          case BigOrdinal => true
+          case BitVector(width: nat) => true
+          case Collection(finite: bool, kind: CollectionKind, eltType: Type) =>
+            && eltType.WellFormed()
+               // This condition is overly restrictive: we will do the general case later.
+               // For instance, maps can contain keys which don't have a decidable equality,
+               // and sequences can contain elements which also don't have a decidable equality
+               // (in which case we don't have a decidable equality over the sequences, but it
+               // is fine).
+            && eltType.NoLeftFunction()
+            && match kind {
+                 case Seq => true
+                 case Set => eltType.NoLeftFunction()
+                 case Multiset => eltType.NoLeftFunction()
+                 case Map(kt) => kt.WellFormed() && kt.NoLeftFunction()
+               }
+          case Function(args: seq<Type>, ret: Type) =>
+            && (forall i | 0 <= i < |args| :: args[i].WellFormed())
+            && ret.WellFormed()
+          case Class(classType: ClassType) =>
+            && (forall i | 0 <= i < |classType.typeArgs| :: classType.typeArgs[i].WellFormed())
         }
       }
-
-      type T(!new,00,==) = Type
     }
 
-    type Type = Types.T
+    type T(!new,00,==) = Type
+  }
+
+  type Type = Types.T
 
-    datatype Tokd<T> =
-      Tokd(tok: Boogie.IToken, val: T)
+  datatype Tokd<T> =
+    Tokd(tok: Boogie.IToken, val: T)
 
-    module BinaryOps {
-      datatype Logical =
-        Iff // And, Or, and Imp are in LazyOp
-      datatype Eq =
+  module BinaryOps {
+    datatype Logical =
+      Iff // And, Or, and Imp are in LazyOp
+    datatype Eq =
         EqCommon | NeqCommon
```

Comment:
2 spaces less indent?
Better would be formatted onto the same line as 'datatype'

Reply:
And less indent for the subsequent declarations also

Reply:
I won't remove newlines in the scope of this PR, but I removed the extra indent in this particular case

Reply:
FIne